### PR TITLE
fix(example/basic): select resizeMode

### PR DESCRIPTION
**Pull request by seyedmostafahasani**
## Summary
Handle resizeMode

### Motivation
Fix a bug with `resizeMode` in the basic app for current testing.

## Test plan

### CODE REVIEW

File: examples/basic/src/components/Overlay.tsx [hunk 1]
```diff
@@ -216,8 +217,17 @@ const _Overlay = forwardRef<VideoRef, Props>((props, ref) => {
...
+        case ResizeMode.CONTAIN:
+          setResizeMode(ResizeMode.CONTAIN);
+          break;
+        case ResizeMode.COVER:
+          setResizeMode(ResizeMode.COVER);
+          break;
+        default:
+          setResizeMode(ResizeMode.STRETCH);
+      }
     }
```
Comment (moskalakamil) [inline examples/basic/src/components/Overlay.tsx]: Is it possible to make setResizeMode(value)? I think switch can be overcoding

Comment (seyedmostafahasani) [reply examples/basic/src/components/Overlay.tsx]: The setResizeMode function accepts `EnumValue<ResizeMode>`, so we cannot pass a string to this function. Instead, we can cast the string to `EnumValue<ResizeMode>`.
`setResizeMode(value as EnumValues<ResizeMode>);`

Comment (moskalakamil) [reply examples/basic/src/components/Overlay.tsx]: maybe we can make some changes to be able to pass the `EnumValue<ResizeMode>` value type to this function?
```tsx
const onResizeModeSelected = (value: EnumValue<ResizeMode>) => {
```

Comment (seyedmostafahasani) [reply examples/basic/src/components/Overlay.tsx]: If you look at the code, you'll see that we have a component called `MultiValueControl` that is used by rate, volume, and resizeMode. Therefore, we cannot change the type of this function.

Comment (moskalakamil) [reply examples/basic/src/components/Overlay.tsx]: Sure! I get it

Comment (moskalakamil) [reply examples/basic/src/components/Overlay.tsx]: I found a way to do this, when replacing `MultiValueControl.tsx` with this code:
```tsx
import React from 'react';
import {
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
} from 'react-native';
import {ResizeMode} from 'react-native-video';

/*
 * MultiValueControl displays a list clickable text view
 */

interface MultiValueControlType<T> {
  // a list a string or number to be displayed
  values: Array<T>;
  // The selected value in values
  selected?: T;
  // callback to press onPress
  onPress: (arg: T) => void;
}

const MultiValueControl = <T extends number | string | ResizeMode>({
  values,
  selected,
  onPress,
}: MultiValueControlType<T>) => {
  const selectedStyle: TextStyle = StyleSheet.flatten([
    styles.option,
    {fontWeight: 'bold'},
  ]);

  const unselectedStyle: TextStyle = StyleSheet.flatten([
    styles.option,
    {fontWeight: 'normal'},
  ]);

  return (
    <View style={styles.container}>
      {values.map(value => {
        const _style = value === selected ? selectedStyle : unselectedStyle;
        return (
          <TouchableOpacity
            key={value}
            onPress={() => {
              onPress?.(value);
            }}>
            <Text style={_style}>{value}</Text>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  option: {
    alignSelf: 'center',
    fontSize: 11,
    color: 'white',
    paddingLeft: 2,
    paddingRight: 2,
    lineHeight: 12,
  },
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default MultiValueControl;
```
And in `Overlay.tsx` change the following:

```tsx
  const onResizeModeSelected = (value: EnumValues<ResizeMode>) => {
  // ...
  const onVolumeSelected = (value: number) => {
  // ...
  const onRateSelected = (value: number) => {
```
We are able to have less code in our app

Comment (seyedmostafahasani) [reply examples/basic/src/components/Overlay.tsx]: Thank you for the suggestion, but I think it isn't related to this PR. We can address it in another PR.

**Review by KrzysztofMoch — APPROVED** [review]

File: examples/basic/src/components/Overlay.tsx [hunk 2]
```diff
@@ -216,8 +217,17 @@ const _Overlay = forwardRef<VideoRef, Props>((props, ref) => {
   };
 
   const onResizeModeSelected = (value: MultiValueControlPropType) => {
-    if (typeof value === 'object') {
-      setResizeMode(value);
+    if (typeof value === 'string') {
+      switch (value) {
+        case ResizeMode.CONTAIN:
```
Comment (freeboub) [inline examples/basic/src/components/Overlay.tsx]: Following previous comment, I think you can use 'as' to simplify this function, can you have a look please?

**Review by moskalakamil — APPROVED** [review]

**Review by freeboub — APPROVED** [review]

**Pull request merged**